### PR TITLE
FOUNDATIONS: Remove Ticket

### LINKED ISSUE #92: FOUNDATIONS: Remove Ticket

### PR DESCRIPTION
**Pull request by StudentofTATU**
Closes #92

### CODE REVIEW

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RemoveById.cs [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+            TicketDependencyValidationException actualTicketDependencyValidationException =
+                await Assert.ThrowsAsync<TicketDependencyValidationException>(
+                    removeTicketByIdTask.AsTask);
+
+            // then
+            actualTicketDependencyValidationException.Should().BeEquivalentTo(
+                expectedTicketDependencyValidationException);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectTicketByIdAsync(It.IsAny<Guid>()),
```
Comment (jamshidbek001) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RemoveById.cs:91]: 2 line

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RemoveById.cs [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedTicketDependencyValidationException))),
```
Comment (jamshidbek001) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RemoveById.cs:96]: 3 line

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RemoveById.cs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.DeleteTicketAsync(It.IsAny<Ticket>()),
+                    Times.Never);
```
Comment (jamshidbek001) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RemoveById.cs:101]: 2 line

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RemoveById.cs [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+                expectedTicketServiceException);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectTicketByIdAsync(It.IsAny<Guid>()),
+                    Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(
+                    expectedTicketServiceException))),
+                        Times.Once);
```
Comment (jamshidbek001) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Exceptions.RemoveById.cs:144]: 3 line

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Validations.RemoveById.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            //then
+            actualTicketValidationException.Should().BeEquivalentTo(expectedTicketValidationException);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(expectedTicketValidationException))),
+                    Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.DeleteTicketAsync(It.IsAny<Ticket>()),
+                    Times.Never);
```
Comment (jamshidbek001) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Validations.RemoveById.cs:51]: 2 line

**Review by jamshidbek001 — CHANGES_REQUESTED** [review]
please correct mistakes

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Validations.RemoveById.cs [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+            TicketValidationException actualTicketValidationException =
+                await Assert.ThrowsAsync<TicketValidationException>(
+                    removeTicketByIdTask.AsTask);
+
+            //then
+            actualTicketValidationException.Should().BeEquivalentTo(expectedTicketValidationException);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(expectedTicketValidationException))),
+                    Times.Once);
```
Comment (StudentofTATU) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Tickets/TicketServiceTests.Validations.RemoveById.cs:47]: this.loggingBrokerMock.Verify(broker =>
    broker.LogError(It.Is(SameExceptionAs(
        expectedTicketValidationException))),Times.Once);

File: Tarteeb.Api/Services/Foundations/Tickets/TicketService.cs [hunk 7]
```diff
@@ -52,5 +52,18 @@ public partial class TicketService : ITicketService
...
             return maybeTicket;
         });
+
+        public ValueTask<Ticket> RemoveTicketByIdAsync(Guid ticketId) =>
+        TryCatch(async () =>
+        {
+            ValidateTicketId(ticketId);
+
+            Ticket maybeTicket = await this.storageBroker
+                 .SelectTicketByIdAsync(ticketId);
```
Comment (StudentofTATU) [inline Tarteeb.Api/Services/Foundations/Tickets/TicketService.cs:62]: delete extra one space

**Review by StudentofTATU — COMMENTED** [review]
Check out

**Pull request merged**